### ISSUE TITLE
windows media player plays recorded video without sound

### ISSUE THREAD

**Issue by Dragon-S** [hughfenghen/WebAV#10]
Run the demo under the av-canvas directory, record the video, and use Windows media player to play without sound. Use other players to play normally (such as: vlc).

The file is as follows:
[WebAv-export-1686045983537.zip](https://github.com/hughfenghen/WebAV/files/11663190/WebAv-export-1686045983537.zip)

**Comment by hughfenghen** [comment]
@Dragon-S 
Windows media player does not support AAC encoded audio by default.

FYI: https://answers.microsoft.com/en-us/windows/forum/all/how-can-i-play-aac-audio-stream-with-windows-media/4e7dfa1e-a166-4238-aacd-8b4d2c17e1f0

<img width="1427" alt="image" src="https://github.com/hughfenghen/WebAV/assets/3307051/129a77d6-53ec-4776-8b3e-46b4fc61438d">

**Comment by hughfenghen** [comment]
Windows media player 12 supports AAC, please confirm the player version.

https://support.microsoft.com/en-us/topic/file-types-supported-by-windows-media-player-32d9998e-dc8f-af54-7ba1-e996f74375d9

**Comment by Dragon-S** [comment]
> Windows media player 12 supports AAC, please confirm the player version.
> 
> https://support.microsoft.com/en-us/topic/file-types-supported-by-windows-media-player-32d9998e-dc8f-af54-7ba1-e996f74375d9

The version of Windows media player I am using is as follows:
<img width="313" alt="屏幕截图 2023-06-07 142549" src="https://github.com/hughfenghen/WebAV/assets/36149358/6fba4588-2b96-4b59-afe9-8b9f2b060389">

**Comment by Dragon-S** [comment]
If I simply convert the WebAv-export-1686045983537.mp4 file with ffmpeg, it can be played. The conversion command is as follows:
`. \ ffmpeg.exe -i C: \ Users \ Lorne \ Downloads \ Webav-EXPORT-1686045983537 \ Webav-EXPORT -686045983537.MP4 C: \ Lorne \ DownnLoads \ Webav-EXPORT-1686045983537 \ Webav-EXPORT-1686045983537- ffmpeg.mp4`


Use ffprobe to view the video information before and after conversion. The audio encoding is different, as follows:
<img width="1281" alt="1" src="https://github.com/hughfenghen/WebAV/assets/36149358/3d0700b9-38c1-4da1-86a1-3c7e2b7b8570">



Is it possible that when the audio encoding is not "aac (LC)" it cannot be played?
However, AudioEncoder has been set to "mp4a.40.2" (MPEG-4 AAC LC) in the code, why only "aac" is displayed in the media information?

**Comment by hughfenghen** [comment]
I tried to solve this over the weekend

**Comment by hughfenghen** [comment]
@Dragon-S 
![image](https://github.com/hughfenghen/WebAV/assets/3307051/1e230ead-e346-4295-9712-fff5a09a6bf0)

I found some clues, maybe missing the esds box, I need more time to verify.

**Comment by hughfenghen** [comment]
https://hughfenghen.github.io/WebAV/demo/record-avcanvas.html

@Dragon-S  Help confirm whether the issues is fixed.
If you are running in dev mode, make sure the mp4box module is updated (there is a bug in mp4box.js building esds box).

![image](https://github.com/hughfenghen/WebAV/assets/3307051/8861b389-d7bc-4fa3-9d71-aba78815f705)

**Comment by Dragon-S** [comment]
> https://hughfenghen.github.io/WebAV/demo/record-avcanvas.html
> 
> @Dragon-S Help confirm whether the issues is fixed. If you are running in dev mode, make sure the mp4box module is updated (there is a bug in mp4box.js building esds box).
> 
> ![image](https://user-images.githubusercontent.com/3307051/245379846-8861b389-d7bc-4fa3-9d71-aba78815f705.png)

Test Environment:
Windows 11 22H2
Chrome 114
Windows Media Player 12

1. I used your demo (https://hughfenghen.github.io/WebAV/demo/record-avcanvas.html) to record a video, but the generated file cannot be played. The recorded file is as follows:
[WebAv-export-1686707123898.zip](https://github.com/hughfenghen/WebAV/files/11740812/WebAv-export-1686707123898.zip)


Analyzed using ffprobe:

<img width="564" alt="屏幕截图 2023-06-14 094803" src="https://github.com/hughfenghen/WebAV/assets/36149358/5128816d-f51d-4905-b688-0ac907d73d95">

Analyzed using mp4box:

<img width="644" alt="屏幕截图 2023-06-14 094924" src="https://github.com/hughfenghen/WebAV/assets/36149358/e8b2bd5b-07d2-4cf7-a295-45f273adf9da">

2. I updated to your latest code (519e68ad147d8c897ea2c48c32f5bb9d60dde2d5) and ran the demo in av-recorder locally. However, the recorded video file still has no audio when played in Windows Media Player. The recorded file is as follows:
[WebAv-export-1686707058135.zip](https://github.com/hughfenghen/WebAV/files/11740821/WebAv-export-1686707058135.zip)


Analyzed using ffprobe:

<img width="576" alt="屏幕截图 2023-06-14 094834" src="https://github.com/hughfenghen/WebAV/assets/36149358/1198c5c7-9cd2-4936-8731-9350ffefe368">

Analyzed using mp4box:

<img width="661" alt="屏幕截图 2023-06-14 095014" src="https://github.com/hughfenghen/WebAV/assets/36149358/6657634e-b200-41ba-ba2c-88b271f109d7">

In summary, the error message is different from before, but the problem still exists.

**Comment by hughfenghen** [comment]
@Dragon-S 
Error log: `overread end of atom 'esds' by 6 bytes` is because of mp4box local cache.
I have updated the version number of hughfenghen/mp4box.js, please make sure it is `mp4box@0.5.3-fenghen`.
![image](https://github.com/hughfenghen/WebAV/assets/3307051/aa596b98-fad8-498a-b56e-7abe2c2fd9e3)

Error log: `Cloud not found codec parameter...`, I didn't reproduce.
Probably the reason is that the moov box was written into the file before it was completed.
I have encountered similar errors: `trun track id unknown, no tfhd was found`, has been resolved here : https://github.com/hughfenghen/WebAV/commit/fac65e47164b0be219602eedf5912d170f740b82

**Comment by Dragon-S** [comment]
@hughfenghen 
Thank you very much! That's great! My Windows Media Player can play the recorded files with sound now. The problem last time was probably due to a cache issue.
